geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/PartitionedRepositoryManager.java

```diff
@@ -105,6 +105,37 @@ public void setUserRegionForRepositoryManager(PartitionedRegion userRegion) {
     return repos;
   }
 
+  @Override
+  public Collection<IndexRepository> getRepositories(RegionFunctionContext ctx, boolean waitOnRetry)
```

Comment:
Would it make sense to have the original getRepositories(ctx) just call this with a false as a second parameter?  This would reduce code duplication I think?

Also, perhaps name the waitOnRetry to "waitForRepository" or "blockUntilReady"